kyaml/yaml/rnode.go

```diff
@@ -248,6 +443,391 @@ func (rn *RNode) IsNilOrEmpty() bool {
+		return 0, true, fmt.Errorf("node was not an int, was %v", scalarNode.value.Tag)
+	}
+}
+
+// GetNestedFloat returns a nested float.
+func (rn *RNode) GetNestedFloat(fields ...string) (float64, bool, error) {
+	scalarNode, found, err := rn.nestedScalar(fields...)
+	if err != nil || !found {
+		return 0, found, err
+	}
+
+	switch scalarNode.value.Tag {
+	case NodeTagFloat:
+		f, err := strconv.ParseFloat(scalarNode.value.Value, 64)
+		if err != nil {
+			return 0, true, fmt.Errorf("unable to parse %q as float: %w", scalarNode.value.Value, err)
+		}
+		return f, true, nil
+	default:
+		return 0, true, fmt.Errorf("node was not a float, was %v", scalarNode.value.Tag)
+	}
+}
+
+// GetNestedBool returns a nested boolean.
+func (rn *RNode) GetNestedBool(fields ...string) (bool, bool, error) {
+	scalarNode, found, err := rn.nestedScalar(fields...)
+	if err != nil || !found {
+		return false, found, err
+	}
+
+	switch scalarNode.value.Tag {
+	case NodeTagBool:
+		b, err := strconv.ParseBool(scalarNode.value.Value)
+		if err != nil {
+			return false, true, fmt.Errorf("unable to parse %q as bool: %w", scalarNode.value.Value, err)
+		}
+		return b, true, nil
+	default:
+		return false, true, fmt.Errorf("node was not a bool, was %v", scalarNode.value.Tag)
+	}
+}
+
+// GetLineComment return the line comment of the field.
+func (rn *RNode) GetLineComment(fields ...string) (string, error) {
+	node, found, err := rn.nestedValue(fields...)
+	if err != nil {
+		return "", err
+	}
+	if !found {
+		return "", fmt.Errorf("unable to get line comment for field %v, since it doesn't exist", fields)
+	}
+	return node.value.LineComment, nil
+}
+
+// GetHeadComment return the head comment of the field.
+func (rn *RNode) GetHeadComment(fields ...string) (string, error) {
+	node, found, err := rn.nestedValue(fields...)
+	if err != nil {
+		return "", err
+	}
+	if !found {
+		return "", fmt.Errorf("unable to get line comment for field %v, since it doesn't exist", fields)
+	}
+	return node.value.HeadComment, nil
+}
+
+// SetOrDie sets a nested field located by fields to the value provided as val.
+// It will panic if it encounters any error.
+func (rn *RNode) SetOrDie(val interface{}, fields ...string) {
+	if err := rn.Set(val, fields...); err != nil {
+		panic(err)
+	}
+}
+
+// Set sets a nested field located by fields to the value provided as val. val
+// can be any type. e.g. int, string, []string, map[string]string, struct like
+// corev1.PodTemplate, yaml.RNode or yaml.Node.
+func (rn *RNode) Set(val interface{}, fields ...string) error {
+	if rn.IsNil() {
+		return ErrRNodeNotFound{}
+	}
+	err := func() error {
+		rnode, err := NewRNodeFrom(val)
+		if err != nil {
+			return err
+		}
+		if rnode == nil {
+			return nil
+		}
+		return rn.setYNode(rnode.YNode(), fields...)
+	}()
+	if err != nil {
+		return fmt.Errorf("unable to set value %#v at fields %v with error: %w", val, fields, err)
+	}
+	return nil
+}
+
+func (rn *RNode) setYNode(yn *Node, fields ...string) error {
```

Comment:
Thinking about the public method `RNode.SetYNode`, would it be beneficial to users to change that one to accept a list of fields? If the user provides no fields, it can do what it currently does; otherwise do what this method does?

Reply:
That's a good point. This one can be merged with existing `SetYNode`.

I don't think we can remove `RNode.SetYNode`, since the low-level API surface is still part of the RNode.